sim/druid/talents.go

```diff
@@ -314,14 +314,15 @@ func (druid *Druid) applyOmenOfClarity() {
 				spell.CostMultiplier += 1
 			}
 		},
-		OnCastComplete: func(aura *core.Aura, sim *core.Simulation, spell *core.Spell) {
```

Comment:
Instead of deleting this, I think we could have a local variable `var proccedAt time.Duration` which is set on proc, and then only remove the aura if `proccedAt != sim.CurrentTime`.

Then you'll be able to remove the hard-coded handling of clearcasting from all the spells.

Reply:
I don't think that would work. The issue here why we have to change anything at all is that we have a CC proc -> we cast something for free -> OnSpellHitDealt is called and may proc another CC -> OnCastComplete is called and removes the CC proc. Weirdly even when moving the check into ApplyEffects the DPS is still different, I have no clue yet why that is.

Thinking about it, maybe we can check in OnCastComplete if the aura was applied at the exact same time as sim.CurrentTime and only remove it if they differ.